### PR TITLE
git-clone: support mounting custom CAs for https

### LINKED ISSUE #660: git-clone tasks should support mounting custom CAs for https git urls
# Expected Behavior

There should be a parameter for attaching a secret with a custom CA.

# Actual Behavior

With https git URLs that use private CAs, I get this error: 

`SSL certificate problem: self signed certificate in certificate chain`

The workaround is to disable SSL verification, but this is dangerous.

# Steps to Reproduce the Problem

1. Create a pipeline using a git-clone task with a https git url using a private CA
2. Run the pipeline
3. Observe the error

# Additional Info

The git-cli could use support for this as well.

### PR DESCRIPTION
**Pull request by chmouel**
Add a optional workspace for custom CAs. This will use pass down the optional
workspace path to git's GIT_SSL_CAPATH env variable (and subsequently to
libcurl).

Fixes #660 

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by chmouel** [comment]
Trying something else to easy the pain to review catalog version bump, I copy the old task version to the new one and make a commit and then add a change and make a commit. Which would make things much easier to review and get clean history.

cc @tektoncd/catalog-collaborators  @afrittoli

**Comment by sbwsg** [comment]
Should this also include an additional `optional` workspace explicitly for the CA certs?  Or is user required to pass them in via `ssh-directory` or `basic-auth` workspaces?

**Comment by chmouel** [comment]
@sbwsg i think this is useful i had someone asking me about it, any chance to review ?

diff is in this commit [`127764f` (#787)](https://github.com/tektoncd/catalog/pull/787/commits/127764fcd1e79b705e2adb338c89b0dfd5764d3d)

**Comment by sbwsg** [comment]
@chmouel sure thing!  How does a user get the certs into the task?  The PR description says "(usually from a bounded workspace path)" but I don't see an additional workspace added to the task.  Is that intentional?

**Comment by sbwsg** [comment]
I also really like the approach for generating the diff. Makes review much easier, cheers!

**Comment by chmouel** [comment]
@sbwsg now using a workspace and removed the params!

**Review by vdemeester — APPROVED** [review]

**Review by vinamra28 — COMMENTED** [review]
a couple of comments. Also can you please squash the commits?

File: task/git-clone/0.5/git-clone.yaml [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+      type: string
+      default: ""
+    - name: verbose
+      description: Log the commands that are executed during `git-clone`'s operation.
+      type: string
+      default: "true"
+    - name: gitInitImage
+      description: The image providing the git-init binary that this Task runs.
+      type: string
+      default: "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.21.0"
```
[the remark this answers is not in the document]
Comment (chmouel) [reply task/git-clone/0.5/git-clone.yaml:101]: I guess we decided some time ago that we only update as needed basis!

**Comment by sbwsg** [comment]
Huge +1 for separating these changes into a copy commit and a modify commit. Makes review so much easier.

/lgtm

**Pull request merged**